src/Explorer/Tabs/Tabs.tsx

```diff
@@ -143,11 +166,13 @@ const CloseButton = ({
   active,
   hovering,
   tabKind,
+  queryCopilotState,
 }: {
   tab: Tab;
   active: boolean;
   hovering: boolean;
   tabKind?: ReactTabKind;
+  queryCopilotState: QueryCopilotState;
```

Comment:
You don't need to pass the state into this function. All you have to do it call `useQueryCopilotState.getState().resetQueryCopilotStates()` on line 186.